### ISSUE TITLE
VA Advisory deficiency

### ISSUE THREAD

**Issue by blchoy** [wmo-im/iwxxm#153]
It was lately noticed that the representation of a volcano in a VA Advisory is less than what is required in Annex 3, including the "State or region" and "Submmit elevation".  This should be fixed in 3.0.0RC3.
![image](https://user-images.githubusercontent.com/30064374/55856404-c898e580-5b9c-11e9-8058-524b08eae7d0.png)

**Comment by blchoy** [comment]
The following is the existing implementation inherited from v2.1.1:

```
<iwxxm:issuingVolcanicAshAdvisoryCentre>
    ...
</iwxxm:issuingVolcanicAshAdvisoryCentre>

<iwxxm:volcano>
    <metce:EruptingVolcano gml:id="uuid.1b9ee023-c87d-4fa0-bf9b-bb1c7215286c">
        <gml:description>FL300 REPORTED</gml:description>
        <metce:name>KARYMSKY 1000-13</metce:name>
        <metce:position>
            <gml:Point gml:id="uuid.687c6e95-460a-4397-a602-ea8483acc3fc" srsName="http://www.opengis.net/def/crs/EPSG/0/4979">
                <gml:pos>54.03 159.27 1536</gml:pos>
            </gml:Point>
        </metce:position>
        <metce:eruptionDate>2008-09-23T01:30:00Z</metce:eruptionDate>
    </metce:EruptingVolcano>
</iwxxm:volcano>

<iwxxm:advisoryNumber>2008/4</iwxxm:advisoryNumber>

<iwxxm:informationSource>MTSAT-1R KVERT KEMSD</iwxxm:informationSource>

<iwxxm:colourCode xlink:href="http://codes.wmo.int/49-2/AviationColourCode/RED"/>
```

The drawback of this implementation is that it lacks "state or region" in TAC, "summit elevation" goes into the gml:pos, and parts of the eruption details is going into a gml:description under metce:EruptingVolcano.  As we don't want to change METCE, a possible implementation will be:

```
<iwxxm:issuingVolcanicAshAdvisoryCentre>
    ...
</iwxxm:issuingVolcanicAshAdvisoryCentre>

<iwxxm:volcano>
    <metce:EruptingVolcano gml:id="uuid.1b9ee023-c87d-4fa0-bf9b-bb1c7215286c">
        <metce:name>KARYMSKY 1000-13</metce:name>
        <metce:position>
            <gml:Point gml:id="uuid.687c6e95-460a-4397-a602-ea8483acc3fc"
                       srsDimension="2"
                       axisLabels="Lat Lon"
                       srsName="http://www.opengis.net/def/crs/EPSG/0/4326">
                <gml:pos>54.03 159.27</gml:pos>
            </gml:Point>
        </metce:position>
        <metce:eruptionDate>2008-09-23T01:30:00Z</metce:eruptionDate>
    </metce:EruptingVolcano>
</iwxxm:volcano>

<iwxxm:stateOrRegion>RUSSIA</iwxxm:stateOrRegion>

<iwxxm:summitElevation>
    <aixm:elevation uom=M">1536</aixm:elevation>
    <aixm:verticalDatum>EGM_96</aixm:verticalDatum>
</iwxxm:summitElevation>

<iwxxm:advisoryNumber>2008/4</iwxxm:advisoryNumber>

<iwxxm:informationSource>MTSAT-1R KVERT KEMSD</iwxxm:informationSource>

<iwxxm:colourCode xlink:href="http://codes.wmo.int/49-2/AviationColourCode/RED"/>

<iwxxm:otherEruptionDetails>FL300 REPORTED</otherEruptionDetails>
```

Views are welcome.

**Comment by mgoberfield** [comment]
Stupid questions follow:

1) What is so abhorrent about using a 3-D CRS in <metce:position> ?
2) If a 3-D CRS is unacceptable, wouldn't adding <metce:summitHeight> and
<metce:stateOrRegion> constitute a bug fix release?

It seems to me that METCE needs to be updated and using IWXXM to fix the
problem is a kludge.
Not trying to be argumentative, just curious as to why a 'better' solution
is closed off to us.

mark

**Comment by jkorosi** [comment]
The issue is UNKNOWN volcano and position. How should be this properly encoded?

**Comment by mgoberfield** [comment]
Hello Jan,

Does

   <metce:name>UNKNOWN</metce:name> OR
   <metce:name>UNNAMED</metce:name>

   <metce:position nilReason="http://www.opengis.net/def/nil/OGC/0/unknown
"/>

suffice for the "UNKNOWN volcano and position" case?

mark

**Comment by jkorosi** [comment]
Hello Mark,

I just reacted to

> 1) What is so abhorrent about using a 3-D CRS in <metce:position> ?

Without Choy's fix, if <metce:position/> is set to nil then it will not be possible to set elevation.

Jan

**Comment by mgoberfield** [comment]
I see.  Thank you, Jan.

**Comment by blchoy** [comment]
METCE is WMO Logical Data Model and is suppose to provide a governed description of weather objects for community use.  However, not even other WMO teams are interested in using it.  Probably because it has too weather objects right now or inconvenient to use because of it structure?  To continue our wish to govern weather objects may be we can follow the foot steps of AIRM where only concepts and logical presentations of them are created for others to make reference to through semantic mappings to demonstrate compliance but at the same time giving flexibility to the developers?

**Comment by blchoy** [comment]
This was fixed in IWXXM 3.0.0RC3.
